src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1285,8 +1287,12 @@ def _process_decln(self, parent, decl, default_visibility,
         '''
         (type_spec, attr_specs, entities) = decl.items
 
+        base_type = None
+        precision = None
+
         # Parse type_spec, currently just 'real', 'double precision',
```

Comment:
This is not true anymore, at least derived types are now also supported. I would remove the comment altogether.

Reply:
Done.